src/protocol/ETHBridge.sol

```diff
@@ -12,41 +13,44 @@ contract ETHBridge is IETHBridge {
     /// Incremental nonce to generate unique deposit IDs.
     uint256 private _globalDepositNonce;
 
-    /// Namespace for the ETH bridge.
-    bytes32 internal constant ETH_BRIDGE_NAMESPACE = keccak256("eth-bridge");
+    ISignalService public immutable signalService;
+
+    /// @dev This is the Anchor on L2 and the Checkpoint Tracker on the L1
+    address public immutable trustedCommitmentPublisher;
+
+    constructor(address _signalService, address _trustedCommitmentPublisher) {
+        require(_signalService != address(0), "Empty signal service");
+        require(_trustedCommitmentPublisher != address(0), "Empty trusted publisher");
+        signalService = ISignalService(_signalService);
+        trustedCommitmentPublisher = _trustedCommitmentPublisher;
+    }
 
     /// @inheritdoc IETHBridge
-    function claimed(bytes32 id) public view virtual returns (bool) {
+    function claimed(bytes32 id) public view returns (bool) {
         return _claimed[id];
     }
 
     /// @inheritdoc IETHBridge
-    function getDepositId(ETHDeposit memory ethDeposit) public view virtual returns (bytes32 id) {
+    function getDepositId(ETHDeposit memory ethDeposit) public pure returns (bytes32 id) {
         return _generateId(ethDeposit);
     }
 
     /// @inheritdoc IETHBridge
-    function deposit(address to, bytes memory data) public payable virtual returns (bytes32 id) {
+    function deposit(address to, bytes memory data) public payable returns (bytes32 id) {
         ETHDeposit memory ethDeposit = ETHDeposit(_globalDepositNonce, msg.sender, to, msg.value, data);
         id = _generateId(ethDeposit);
         unchecked {
             ++_globalDepositNonce;
         }
+        signalService.sendSignal(id);
         emit DepositMade(id, ethDeposit);
     }
 
     /// @inheritdoc IETHBridge
-    function claimDeposit(ETHDeposit memory ethDeposit, uint256 height, bytes memory proof)
-        external
-        returns (bytes32 id)
-    {
-        id = _generateId(ethDeposit);
-    }
-
-    /// @dev Processes deposit claim by id.
-    /// @param id Identifier of the deposit
-    /// @param ethDeposit Deposit to process
-    function _processClaimDepositWithId(bytes32 id, ETHDeposit memory ethDeposit) internal virtual {
+    // TODO: Non reentrant
+    function claimDeposit(ETHDeposit memory ethDeposit, uint256 height, bytes memory proof) external {
+        bytes32 id = _generateId(ethDeposit);
+        signalService.verifySignal(height, trustedCommitmentPublisher, ethDeposit.from, id, proof);
```

Comment:
This is a bug. The `sender` should be the source `ETHBridge` contract, not `ethDeposit.from`